openedx_tagging/core/tagging/models/system_defined.py

```diff
@@ -288,17 +233,49 @@ def _get_available_languages(cls) -> set[str]:
             langs.add(django_lang[0].split("-")[0])
         return langs
 
-    def _check_valid_language(self, object_tag: ObjectTag) -> bool:
+    def validate_value(self, value: str):
         """
-        Returns True if the tag is on the available languages
+        Check if 'value' is part of this Taxonomy, based on the specified model.
         """
-        available_langs = self._get_available_languages()
-        if not object_tag.tag:
-            raise AttributeError("Expected object_tag.tag to be set")
-        return object_tag.tag.external_id in available_langs
+        for _, lang_name in settings.LANGUAGES:
+            if lang_name == value:
+                return True
+        return False
+
+    def tag_for_value(self, value: str):
+        """
+        Get the Tag object for the given value.
+        """
+        for lang_code, lang_name in settings.LANGUAGES:
+            if lang_name == value:
+                return self.tag_for_external_id(lang_code)
+        raise Tag.DoesNotExist
 
-    def _check_tag(self, object_tag: ObjectTag) -> bool:
+    def validate_external_id(self, external_id: str):
         """
-        Returns True if the tag is on the available languages
+        Check if 'external_id' is part of this Taxonomy.
+        """
+        lang_code = external_id.lower()
+        LANGUAGE_DICT = getattr(settings, "LANGUAGE_DICT", dict(settings.LANGUAGES))  # setting may or may not exist.
```

Comment:
Oh, I guess `settings.LANGUAGE_DICT` is an old name for `settings.LANGUAGES`? Didn't know that.

But if we need to check both of them, then don't we need to check them both [here](https://github.com/openedx/openedx-learning/pull/87/files#diff-8640a8393169a45e9926177fd6f2e8c7b0ad8439c96554334b95bbe912f52203R240) and [here](https://github.com/openedx/openedx-learning/pull/87/files#diff-8640a8393169a45e9926177fd6f2e8c7b0ad8439c96554334b95bbe912f52203R249) too?

Reply:
Close... Actually settings.LANGUAGES is the normal setting, but it's a list of tuples. For convenience/performance, the `cms`/`lms` also [define](https://github.com/openedx/edx-platform/blob/d60cdc2305fc7ebb99fbffa5fad535a91f9f0be2/cms/envs/common.py#L1227) `LANGUAGE_DICT = dict(LANGUAGES)` which converts it into a dict.

I was already on the fence about whether to use LANGUAGE_DICT or not, but your comment shows to me that it's certainly not clear as it is. So I will add some comments in to clarify before merging.